---
title: "HeatMapsForGreg"
output: html_document
date: "2024-07-03"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


#Generating Heat Maps of only differentially expressed Genes


Loading in read count data and then only selecting those that are DE
```{r}
library(readxl)
library(dplyr)

ReadCount<-read_xlsx("./ParvinTranscriptomic.xlsx")
SCVI<-read_xlsx("./SCVI_SFvsSG_Separated.xlsx")
IMR<-read_xlsx("./IMR90_IFvsIG_Separated.xlsx")

SCVI<-subset(SCVI,adj.P.Val<0.05)

IMR<-subset(IMR,adj.P.Val<0.05)

ReadCountIMR<-ReadCount[,c(14,5:7,2:4)]
ReadCountSCVI<-ReadCount[,c(14,11:13,8:10)]

SigIMRGenes<-na.omit(IMR$GeneSymbol)
SigSCVIGenes<-na.omit(SCVI$GeneSymbol)

```

```{r}
#Normalization and Filtering of IMR
d0IMR <- DGEList(ReadCountIMR)
d0IMR <- calcNormFactors(d0IMR)

cutoff <- 1
drop <- which(apply(cpm(d0IMR), 1, max) < cutoff)
dIMR <- d0IMR[-drop,] 
dim(dIMR)

I_group_labels<- c("IG","IG","IG","IF","IF","IF")
I_groups<-factor(I_group_labels)

mmIMR <- model.matrix(~0 + I_groups)

tmpIMR <- voom(d0IMR, mmIMR, plot = T)

normalizedIMR<-tmpIMR$E
#generates a normalized read count matrix
```

```{r}
fullNormalizedIMR<-normalizedIMR
```

```{r}
#filtering normalized reads for only those Sig Genes
row.names(normalizedIMR)<-tmpIMR$genes$GeneSymbol


normalizedIMR<- as.data.frame(normalizedIMR) %>%
  filter(rownames(as.data.frame(normalizedIMR)) %in% SigIMRGenes)
```

```{r}
#computing median expression and log2FC
median_expressionIMR <- rowMedians(as.matrix(normalizedIMR))
logFC_IMR <- log2(normalizedIMR / median_expressionIMR)

logFC_IMR[is.nan(as.matrix(logFC_IMR))] <- 0
```

```{r}
heatmap(as.matrix(logFC_IMR[1:30,]), 
        scale="row", 
        col=colorRampPalette(c("blue", "white", "red"))(100), 
        margins=c(5, 5), 
        cexRow=0.8, 
        cexCol=0.8)

```


```{r}
heatmap(as.matrix(logFC_IMR))
```

```{r}
library(pheatmap)
pheatmap(logFC_IMR[1:50,],
         color=colorRampPalette(c("blue", "white", "red"))(100),
         scale = "row", 
         clustering_distance_rows = "euclidean", 
         clustering_distance_cols = "euclidean", 
         clustering_method = "complete", 
         show_rownames = TRUE, 
         show_colnames = TRUE,
         fontsize=5,
         main = "Heatmap")
```


```{r}
save_pheatmap_pdf <- function(x, filename, width=7, height=7) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}
```
```{r}
saveThis<-pheatmap(logFC_IMR,
         color=colorRampPalette(c("blue", "white", "red"))(100),
         scale = "row", 
         clustering_distance_rows = "euclidean", 
         clustering_distance_cols = "euclidean", 
         clustering_method = "complete", 
         show_rownames = TRUE, 
         show_colnames = TRUE,
         fontsize=10,
         main = "Heatmap of Log2 Fold Changes from Median Expression")

save_pheatmap_pdf(saveThis, "./IMR_HeatMap.pdf")
```


## 7/8 Heatmaps for Parvin

Heatmap of Top 25 with Lowest Adj.p.Val

```{r}


# Select the top 25 rows
IMRTop25 <- head(IMR, 26)
IMRTop25Genes<-IMRTop25$GeneSymbol
IMRTop25Genes<-na.omit(IMRTop25Genes)
IMRTop25Genes<-c(IMRTop25Genes,"CYMP")


#Filter based on these genes 
Top25IMRLogFC<- as.data.frame(logFC_IMR) %>%
  filter(rownames(as.data.frame(logFC_IMR)) %in% IMRTop25Genes)
```

```{r}
saveThis<-pheatmap(Top25IMRLogFC,
         color=colorRampPalette(c("blue", "white", "red"))(100),
         scale = "row", 
         clustering_distance_rows = "euclidean", 
         clustering_distance_cols = "euclidean", 
         clustering_method = "complete", 
         show_rownames = TRUE, 
         show_colnames = TRUE,
         fontsize=10,
         main = "Heatmap of Log2 Fold Changes from Normalized Median Expression")

save_pheatmap_pdf(saveThis, "./IMR_Top25.pdf")
```

# 7/8/24 PCA

```{r}
plotpca <- function(df, seq, color, fn, p){ 
  # df = data.frame that acts as the input for the function.
  # seq = sequencing info housing the categories that each sample belongs to.
  # color = colors to be used for PCA plot.
  # fn = filename used for savin the plot.
  # p = path for the file.
  pca_rna <- prcomp(t(df), scale = TRUE, center = TRUE)
  per_var <- 100*(pca_rna$sdev^2/sum(pca_rna$sdev^2))
  d <- data.frame("PC1" = pca_rna$x[,1],
                  "PC2" = pca_rna$x[,2],
                  "PC3" = pca_rna$x[,3],
                  "PC4" = pca_rna$x[,4],
                  "Condition" = seq)
  g <- ggplot(d, aes(x = PC1, y = PC2)) +
          geom_point(aes(color = Condition),
                     size=5,
                     stroke = 0.8
          ) +
          scale_shape_manual(values = c(15, 17)
          ) +
          scale_color_manual(values = color
          ) +
          labs(
            x = paste0("PC1, VarExp: ", round(per_var[[1]], 1), "%"),
            y = paste0("PC2, VarExp: ", round(per_var[[2]], 1), "%"),
            color = guide_legend(title = "Condition", size = 18)
          ) +
          theme(
            panel.background = element_blank(),
            panel.grid.minor = element_blank(),
            panel.grid.major = element_blank(),
            text = element_text(size = 18),
            axis.text.x = element_text(size = 16),
            axis.text.y = element_text(size = 16),
            panel.grid = element_line(color = "gray0", linetype = "solid"),
            legend.key = element_blank(),
            legend.text = element_text(size = 16),
            legend.title = element_text(face = "italic")
          ) +
          guides(
            shape = guide_legend(override.aes = list(size = 4)), # Making legend shapes smaller
            color = guide_legend(override.aes = list(size = 4)), # Making legend colors smaller
          ) +
          geom_hline(aes(yintercept = 0), 
                     color = "gray0") +
          geom_vline(aes(xintercept = 0),
                     color = "gray0")
  ggsave(paste0(fn, "PC1_vs_PC2.tiff"), plot = g, device = "tiff", path = p)
}
```

```{r}
library(ggplot2)
GroupsIMR<-factor(c("Ground","Ground","Ground","Flight","Flight","Flight"))

plotpca(as.matrix(fullNormalizedIMR), GroupsIMR, color = c("#ec696f", "#53dbf3"), fn = "IMR_PCA", p = "./PCA")

```

```{r}
g<-ggplot(d, aes(x = PC1, y = PC2, color = Condition)) +
  geom_point(size = 3) +
  labs(
    x = paste0("PC1, VarExp: ", round(per_var[[1]], 1), "%"),
    y = paste0("PC2, VarExp: ", round(per_var[[2]], 1), "%"),
    title = "PCA"
  ) +
  theme_minimal()+
  scale_color_manual(values = c("#ec696f", "#53dbf3"))


```
```{r}
ggsave("IMR_PCA_PC1vsPC2.tiff", plot = g, device = "tiff", path = "./PCA")

```


```{r}
library(edgeR)
library(readr)
library(dplyr)

df<-fullNormalizedIMR
seq<-GroupsIMR
color = c("#ec696f", "#53dbf3")

  pca_rna <- prcomp(t(df), scale = TRUE, center = TRUE)
  per_var <- 100*(pca_rna$sdev^2/sum(pca_rna$sdev^2))
  d <- data.frame("PC1" = pca_rna$x[,1],
                  "PC2" = pca_rna$x[,2],
                  "PC3" = pca_rna$x[,3],
                  "PC4" = pca_rna$x[,4],
                  "Condition" = seq)
  g <- ggplot(d, aes(x = PC1, y = PC2)) +
          geom_point(aes(color = Condition,
                         shape=Condition),
                     size=5,
                     stroke = 0.8
          ) +
          scale_shape_manual(values = c(15, 17)
          ) +
          scale_color_manual(values = color
          ) +
          labs(
            x = paste0("PC1, VarExp: ", round(per_var[[1]], 1), "%"),
            y = paste0("PC2, VarExp: ", round(per_var[[2]], 1), "%"),
            color = guide_legend(title = "Condition", size = 18)
          ) +
          theme(
            panel.background = element_blank(),
            panel.grid.minor = element_blank(),
            panel.grid.major = element_blank(),
            text = element_text(size = 18),
            axis.text.x = element_text(size = 16),
            axis.text.y = element_text(size = 16),
            panel.grid = element_line(color = "gray0", linetype = "solid"),
            legend.key = element_blank(),
            legend.text = element_text(size = 16),
            legend.title = element_text(face = "italic")
          ) +
          guides(
            shape = guide_legend(override.aes = list(size = 4)), # Making legend shapes smaller
            color = guide_legend(override.aes = list(size = 4)), # Making legend colors smaller
          ) +
          geom_hline(aes(yintercept = 0), 
                     color = "gray0") +
          geom_vline(aes(xintercept = 0),
                     color = "gray0")
  
```
```{r}
g
```

```{r}
ggsave("", plot = g, device = "tiff", path = p)

```

# 7/15/2024 SCVI Heatmaps and PCA

##Heat MAP
Heatmap of Top 25 with Lowest Adj.p.Val

```{r}
#Normalization and Filtering of SCVI
d0SCVI <- DGEList(ReadCountSCVI)
d0SCVI <- calcNormFactors(d0SCVI)

cutoff <- 1
drop <- which(apply(cpm(d0SCVI), 1, max) < cutoff)
dSCVI <- d0SCVI[-drop,] 
dim(dSCVI)

S_group_labels<- c("SG","SG","SG","SF","SF","SF")
S_groups<-factor(S_group_labels)

mmSCVI <- model.matrix(~0 + S_groups)

tmpSCVI <- voom(d0SCVI, mmSCVI, plot = T)

normalizedSCVI<-tmpSCVI$E
#generates a normalized read count matrix
```

```{r}
fullNormalizedSCVI<-normalizedSCVI
```

```{r}
#filtering normalized reads for only those Sig Genes
row.names(normalizedSCVI)<-tmpSCVI$genes$GeneSymbol


normalizedSCVI<- as.data.frame(normalizedSCVI) %>%
  filter(rownames(as.data.frame(normalizedSCVI)) %in% SigSCVIGenes)
```

```{r}
#computing median expression and log2FC
median_expressionSCVI <- rowMedians(as.matrix(normalizedSCVI))
logFC_SCVI <- log2(normalizedSCVI / median_expressionSCVI)

logFC_SCVI[is.nan(as.matrix(logFC_SCVI))] <- 0
```


```{r}


# Select the top 25 rows
SCVITop25 <- head(SCVI, 25)
SCVITop25Genes<-SCVITop25$GeneSymbol


#Filter based on these genes 
Top25SCVILogFC<- as.data.frame(logFC_SCVI) %>%
  filter(rownames(as.data.frame(logFC_SCVI)) %in% SCVITop25Genes)
```

```{r}
saveThis<-pheatmap(Top25SCVILogFC,
         color=colorRampPalette(c("blue", "white", "red"))(100),
         scale = "row", 
         clustering_distance_rows = "euclidean", 
         clustering_distance_cols = "euclidean", 
         clustering_method = "complete", 
         show_rownames = TRUE, 
         show_colnames = TRUE,
         fontsize=10,
         main = "Heatmap of Log2 Fold Changes from Normalized Median Expression")

save_pheatmap_pdf(saveThis, "./SCVI_Top25.pdf")
```
## PCA
```{r}
GroupsSCVI<-factor(c("Ground","Ground","Ground","Flight","Flight","Flight"))

df<-fullNormalizedSCVI
seq<-GroupsSCVI
color = c("#ec696f", "#53dbf3")

pca_rna <- prcomp(t(df), scale = TRUE, center = TRUE)
per_var <- 100*(pca_rna$sdev^2/sum(pca_rna$sdev^2))
d <- data.frame("PC1" = pca_rna$x[,1],
                "PC2" = pca_rna$x[,2],
                "PC3" = pca_rna$x[,3],
                "PC4" = pca_rna$x[,4],
                "Condition" = seq)
```

```{r}
g<-ggplot(d, aes(x = PC1, y = PC2, color = Condition)) +
  geom_point(size = 3) +
  labs(
    x = paste0("PC1, VarExp: ", round(per_var[[1]], 1), "%"),
    y = paste0("PC2, VarExp: ", round(per_var[[2]], 1), "%"),
    title = "PCA"
  ) +
  theme_minimal()+
  scale_color_manual(values = c("#ec696f", "#53dbf3"))


```
```{r}
ggsave("SCVI_PCA_PC1vsPC2.tiff", plot = g, device = "tiff", path = "./PCA")

```
